flexmeasures/utils/time_utils.py

```diff
@@ -91,11 +91,14 @@ def localized_datetime_str(dt: datetime, dt_format: str = "%Y-%m-%d %I:%M %p") -
     return local_dt.strftime(dt_format)
 
 
-def naturalized_datetime_str(dt: datetime | None, now: datetime | None = None) -> str:
+def naturalized_datetime_str(
+    dt: datetime | str | None, now: datetime | None = None
+) -> str:
     """
     Naturalise a datetime object (into a human-friendly string).
     The dt parameter (as well as the now parameter if you use it)
     can be either naive or tz-aware. We assume UTC in the naive case.
+    If dt parameter is a string it is expected to look like 'Sun, 28 Apr 2024 08:55:58 GMT'.
```

Comment:
I'm confused why this is needed now. We've used `naturalized_datetime_str` from templates in a few other places, see below. Do you know what is different here?

```
± grep -r '| naturalized_datetime' flexmeasures
flexmeasures/ui/templates/views/status.html:                                <span title="{{ sensor['staleness_since'] }}">{{ sensor["staleness_since"] | naturalized_datetime }}</span>
flexmeasures/ui/templates/views/status.html:                                {{ job_data.enqueued_at | naturalized_datetime }}
flexmeasures/ui/templates/views/status.html:                                {{ job_data.enqueued_at | naturalized_datetime }}
flexmeasures/ui/templates/admin/logged_in_user.html:                                {{ logged_in_user.last_login_at | naturalized_datetime }}
flexmeasures/ui/templates/admin/logged_in_user.html:                                {{ logged_in_user.last_seen_at | naturalized_datetime }}
flexmeasures/ui/templates/crud/account.html:                                {{ user.last_login_at | naturalized_datetime}}
flexmeasures/ui/templates/crud/account.html:                                {{ user.last_seen_at | naturalized_datetime}}
flexmeasures/ui/templates/crud/users.html:                                {{ user.last_login_at | naturalized_datetime}}
flexmeasures/ui/templates/crud/users.html:                                {{ user.last_seen_at | naturalized_datetime}}
flexmeasures/ui/templates/crud/user.html:                                    {{  user.last_login_at | naturalized_datetime }}
flexmeasures/ui/templates/crud/user.html:                                    {{  user.last_seen_at | naturalized_datetime }}
```

Reply:
In any case, I would prefer if we'd be able to handle iso8861 dates here (e.g. `2024-04-28T08:55:58`). The format you listed here seems not so widely usable.

Reply:
This is needed here because we make json from internal api response when in other cases we have objects. Happens here: https://github.com/FlexMeasures/flexmeasures/pull/1042/files#diff-87df9c99219fad585ad90d58f6186b2edab5735d4c1e5dfb5dc549a468d93ecaR172. 
Decided to leave this as is but add comment about why a particular format was chosen